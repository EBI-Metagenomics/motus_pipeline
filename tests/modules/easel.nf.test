nextflow_process {

    name "Test module easel"
    script "modules/easel.nf"
    process "EASEL_EXTRACT_BY_COORD"

    test("easel") {

        when {
            params {
                matches = "${baseDir}/tests/modules/fixtures/cmsearch/output/mgtest_merged.fastq.cmsearch_matches.tbl.deoverlapped"
                seqs = "${baseDir}/tests/modules/fixtures/bwa/output/mgtest_merged.fastq.fasta"
            }
            process {
                """
                input[0] = channel.fromPath("${params.seqs}", checkIfExists: true)
                input[1] = channel.fromPath("${params.matches}", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}