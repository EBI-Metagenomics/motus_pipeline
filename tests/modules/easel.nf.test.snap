{
    "easel": {
        "content": [
            {
                "models_fasta": [
                    "mgtest_merged.fastq_mgtest_merged.fastq.cmsearch_matches.tbl.fasta:md5,9b18ab968e855f663e2a7be8ed6097d9"
                ]
            }
        ],
        "timestamp": "2023-03-14T14:39:23+0000"
    }
}